### ISSUE TITLE
Is there a way to get transparent background for segments?

### ISSUE THREAD

**Issue by cannonstar** [bhilburn/powerlevel9k#240]
My terminal has a transparent background so I'd like to have one or two segment backgrounds to be transparent while keeping the foreground a solid color.

**Comment by dritter** [comment]
Well. It would kinda work if you set the background for the segments to an invalid value like `POWERLEVEL9K_TIME_BACKGROUND="none"`.
Kind of, because it would look different from what you want (at least, if you don't use `flat` mode). To go a bit into detail here: The separating glyph (a.k.a. `POWERLEVEL9K_LEFT_SEGMENT_SEPARATOR`) is a real character, which has the same foreground color as the preceding segments background color. That is how we achieve the effect of the segments.

Long story short: If you "disable" the background color of a segment, the separator will still be printed in some color (at least in the terminal emulators default color).

**Comment by cannonstar** [comment]
Ahh I see, probably better to stick with a colored background then. Thanks!

**Comment by giogziro95** [comment]
Super noob question here: Is there a way to make separator character's foreground transparent as well?

**Comment by giogziro95** [comment]
@dritter Super noob question here: Is there a way to make separator character's foreground transparent as well?

**Comment by dritter** [comment]
@giogziro95 Short answer: unfortunately not. 

Transparency is a feature of your Terminal Emulator. So, from the point of view of your Terminal Emulator all you get from P9K/ZSH is the advice to print a character in a specific color (or to print a background color). But it cannot distinguish between the separator and normal characters. This is because the separator is a normal character and not printed in the background. In ZSH you just can send a control character to start or stop a background color. The Terminal Emulator just starts from that point the background color in the line height, which results in a rectangular shape. To create the illusion of that segments we print the separator in the same background color..

**Comment by giogziro95** [comment]
@dritter Thanks for your response. And I'm not that noob; I know about escape codes and how Zsh is doing the tricks. What I was asking was if there was a way to print a character that's transparent/invisible. Obviously, this depends on the terminal emulator as well, and I assumed that there was no way before even finding this issue, but I decided to try anyway. So, apparently, the answer is no, and that's what I was expecting.

Another solution could be not printing the separator character at all, but that would probably require messing with the theme scripts if you need to do this only in one segment. Am I wrong?

**Comment by dritter** [comment]
@giogziro95 You could deactivate the separators at all, just set `POWERLEVEL9K_LEFT_SEGMENT_SEPARATOR=''` ;)

**Comment by giogziro95** [comment]
@dritter Yeah, I know, but that would deactivate all separators as well. 😂

**Comment by dritter** [comment]
:D There is no easy way to disable the separator for a specific segment. Sorry.

**Comment by rjorgenson** [comment]
Depending on the circumstances you could use a joined segment to not have a separator show up. I use this for a couple segments I want to appear 'together' - though I don't know of anything that will allow you to have transparent characters in a terminal.

**Comment by dritter** [comment]
Ha! Right @rjorgenson That is an easy way to accomplish this.. 👍

**Comment by giogziro95** [comment]
@rjorgenson Thanks. However, that wouldn't work either since my goal was to remove this very last (well, the one on the left) character which can't be done through that. Anyway, not a dealbreaker. I can live without that. 🙂

**Comment by giogziro95** [comment]
@rjorgenson Thanks. However, that wouldn't work either since my goal was to remove this very last (well, the one on the left) character which can't be done through that. Anyway, not a dealbreaker. I can live without that. 🙂

![Prompt](https://user-images.githubusercontent.com/2781117/28392939-4ffc3ad8-6cf5-11e7-870b-ca329151a220.png)

**Comment by rjorgenson** [comment]
@giogziro95 you're a far stronger person than I am, that would drive me up the wall =]

**Comment by giogziro95** [comment]
@rjorgenson Haha. Nope, you misunderstood. I just gave it a nice lite grey background. That should do it for now.

![Prompt](https://user-images.githubusercontent.com/2781117/28393645-021ef31a-6cf9-11e7-81f9-2269410c6ea2.png)
